pkg/accountmanager/poller.go

```diff
@@ -165,20 +205,52 @@ func (p *accountPoller) doAccountPolling() {
 	if err := p.inventory.BuildVpcCache(vpcMap, p.namespacedName); err != nil {
 		p.log.Error(err, "failed to build vpc cache", "account", p.namespacedName.String())
 	}
+	p.log.Info("Discovered compute resources statistics", "account", p.namespacedName,
+		"vpcs", len(vpcMap))
 
 	// Perform VM Operations only when CES is added.
 	vmCount := 0
-	if p.selector != nil {
+
+	selectorCache := p.selectorIndexer.List()
+	if len(selectorCache) != 0 {
 		// TODO: Avoid calling plugin to get VM inventory from snapshot.
-		virtualMachines := p.getComputeResources(p.cloudInterface)
-		// TODO: We are walking thru virtual map twice. Once here and second one in BuildVmCAche.
-		// May be expose Add, Delete, Update routine in inventory and we do the calculation here.
-		p.updateAgentState(virtualMachines)
-		p.inventory.BuildVmCache(virtualMachines, p.namespacedName)
-		vmCount = len(virtualMachines)
+		selectorCache := p.getSelectorsForInventoryUpdate()
```

Comment:
why introduce this change variable.. keep it simple.. make new function from 219 to get VM inventory and update VM inventory in the store.. Call all selector from here and whenever a new CES is added, just all new function you have introduced with just 1 selector.. Change variable is overkill IMO